pkgs/development/web/woff2/brotli-static.patch

```diff
@@ -0,0 +1,36 @@
+diff a/cmake/FindBrotliDec.cmake b/cmake/FindBrotliDec.cmake
+--- a/cmake/FindBrotliDec.cmake
++++ b/cmake/FindBrotliDec.cmake
+@@ -18,10 +18,10 @@ find_path(BROTLIDEC_INCLUDE_DIRS
+     HINTS ${PC_BROTLIDEC_INCLUDEDIR}
+ )
+ 
+-find_library(BROTLIDEC_LIBRARIES
+-    NAMES brotlidec
+-    HINTS ${PC_BROTLIDEC_LIBDIR}
+-)
++if(NOT BUILD_SHARED_LIBS)
++  set(_S "STATIC_")
++endif()
++set(BROTLIDEC_LIBRARIES ${PC_BROTLIDEC_${_S}LIBRARIES})
```

Comment:
It seems like the only reason why we need this is because we are applying google/brotli#655 which is not upstream yet. The problem is that other than fixing BUILD_SHARED_LIBS it also sneakily does this:
```
-target_link_libraries(brotlidec-static brotlicommon-static)
-target_link_libraries(brotlienc-static brotlicommon-static)
-
```

I think the 7ebd5999bd11 needs to be reworked to not use unsupported patch. Instead, we could remove the shared libraries and symlink the static ones in their place (also suggested in https://github.com/google/brotli/pull/655#issuecomment-376515181).
cc @matthewbauer @domenkozar